hugr-core/src/export.rs

```diff
@@ -46,6 +46,8 @@ struct Context<'a> {
     term_map: FxHashMap<model::Term<'a>, model::TermId>,
     /// The current scope for local variables.
     local_scope: Option<model::NodeId>,
+    /// Constraints to be added to the local scope.
```

Comment:
Is `local scope` == `[Self::local_scope]`? Do we actually plan to modify the local scope by adding these? Or are these "Constraints to be added to those from the local scope" (i.e., extra constraints from somewhere else)? "Constraints in addition to those from the local scope" would be clearer still that we don't intend to mutate some list.

Reply:
When exporting a `PolyFuncType`, we gather constraints in `local_constraints`. `PolyFuncType` itself can not express constraints, but if one of the parameters takes a copyable type, `local_constraints` is where we record the corresponding constraint for that parameter.